device_backends/LogicalNameMapping/src/LNMBitRangeAccessPlugin.cc

```diff
@@ -99,32 +101,29 @@ namespace ChimeraTK::LNMBackend {
       auto unlock = cppext::finally([this] { this->_lock.unlock(); });
       _target->postRead(type, hasNewData);
       if(!hasNewData) return;
-      auto validity = _target->dataValidity();
 
-      if constexpr(!std::is_integral<UserType>::value) {
-        // This should never be reached, the decorate function should make sure that we only have integral types here
-        assert(false);
-      }
-      else {
-        uint64_t v{};
-        static_assert(sizeof(TargetType) <= sizeof(uint64_t), "Target data type too big.");
-
-        auto targetValue = _target->accessData(0);
-        memcpy(std::addressof(v), std::addressof(targetValue), sizeof(TargetType));
+      if constexpr(std::is_same_v<uint64_t, TargetType>) {
+        auto validity = _target->dataValidity();
+        uint64_t v{_target->accessData(0)};
         v = (v & _maskOnTarget) >> _shift;
 
-        // There are bits set outside of the range of the UserType
-        // Clamping according to B.2.4 and setting the faulty flag
-        // FIXME: Probably easier once the FixpointConverter is supporting 64bit raw types
-        if((v & ~_userTypeMask) != 0) {
-          v = std::numeric_limits<UserType>::max();
+        buffer_2D[0][0] = fixedPointConverter.scalarToCooked<UserType>(uint32_t(v));
+        // Do a quick check if the fixed point converter clamped. Then set the
+        // data validity faulty according to B.2.4.1
+        // For proper implementation of this, the fixed point converter needs to signalize
+        // that it had clamped. See https://redmine.msktools.desy.de/issues/12912
+        auto raw = fixedPointConverter.toRaw(buffer_2D[0][0]);
+        if(raw != v) {
           validity = DataValidity::faulty;
         }
-        memcpy(buffer_2D[0].data(), &v, sizeof(UserType));
-      }
 
-      this->_versionNumber = std::max(this->_versionNumber, _target->getVersionNumber());
-      this->_dataValidity = validity;
+        this->_versionNumber = std::max(this->_versionNumber, _target->getVersionNumber());
+        this->_dataValidity = validity;
+      }
+      else {
+        // This code should never be reached so long the the bit range plugin requires its target type to be uint64
+        assert(false);
```

Comment:
No, we cannot since we get all variants of the function nevertheless, and the static_assert will trigger.  We also cannot use partial specialisation since the template is on the class, not the function.